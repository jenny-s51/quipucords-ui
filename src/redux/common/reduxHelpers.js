import _get from 'lodash/get';
import { helpers } from '../../common/helpers';

const FULFILLED_ACTION = base => `${base}_FULFILLED`;

const PENDING_ACTION = base => `${base}_PENDING`;

const REJECTED_ACTION = base => `${base}_REJECTED`;

const setStateProp = (prop, data, options) => {
  const { state = {}, initialState = {}, reset = true } = options;
  let obj = { ...state };

  if (process.env.REACT_APP_ENV === 'development' && prop && !state[prop]) {
    console.error(`Error: Property ${prop} does not exist within the passed state.`, state);
  }

  if (process.env.REACT_APP_ENV === 'development' && reset && prop && !initialState[prop]) {
    console.warn(`Warning: Property ${prop} does not exist within the passed initialState.`, initialState);
  }

  if (reset && prop) {
    obj[prop] = {
      ...state[prop],
      ...initialState[prop],
      ...data
    };
  } else if (reset && !prop) {
    obj = {
      ...state,
      ...initialState,
      ...data
    };
  } else if (prop) {
    obj[prop] = {
      ...state[prop],
      ...data
    };
  } else {
    obj = {
      ...state,
      ...data
    };
  }

  return obj;
};

const generatedPromiseActionReducer = (types = [], state = {}, action = {}) => {
  const { type } = action;
  const expandedTypes = [];

  types.forEach(
    val =>
      (Array.isArray(val.type) && val.type.forEach(subVal => expandedTypes.push({ ref: val.ref, type: subVal }))) ||
      expandedTypes.push(val)
  );

  const [whichType] = expandedTypes.filter(val =>
    new RegExp(
      `^(${REJECTED_ACTION(val.type || val)}|${PENDING_ACTION(val.type || val)}|${FULFILLED_ACTION(val.type || val)})$`
    ).test(type)
  );

  if (!whichType) {
    return state;
  }

  const baseState = {
    error: false,
    errorMessage: '',
    fulfilled: false,
    metaData: action.meta && action.meta.data,
    metaId: action.meta && action.meta.id,
    metaQuery: action.meta && action.meta.query,
    pending: false,
    update: false
  };

  // Automatically apply data and state to a contextual ID if meta.id exists.
  const idUse = data => {
    const typeId = typeof action?.meta?.id;
    return (
      ((typeId === 'string' || typeId === 'number') &&
        action?.meta?.id && { [action.meta.id]: { ...baseState, ...data } }) || {
        ...baseState,
        ...data
      }
    );
  };

  switch (type) {
    case REJECTED_ACTION(whichType.type || whichType):
      return setStateProp(
        whichType.ref || null,
        idUse({
          error: true,
          errorMessage: helpers.getMessageFromResults(action.payload).message,
          errorStatus: helpers.getStatusFromResults(action.payload)
        }),
        {
          state
        }
      );
    case PENDING_ACTION(whichType.type || whichType):
      return setStateProp(
        whichType.ref || null,
        idUse({
          pending: true
        }),
        {
          state
        }
      );

    case FULFILLED_ACTION(whichType.type || whichType):
      return setStateProp(
        whichType.ref || null,
        idUse({
          date: _get(
            action.payload,
            'headers.date',
            (helpers.DEV_MODE && helpers.getCurrentDate().toUTCString()) || undefined
          ),
          data: (action.payload && action.payload.data) || {},
          fulfilled: true
        }),
        {
          state
        }
      );

    default:
      return state;
  }
};

const reduxHelpers = {
  FULFILLED_ACTION,
  PENDING_ACTION,
  REJECTED_ACTION,
  generatedPromiseActionReducer,
  setStateProp
};

export { reduxHelpers as default, reduxHelpers };
