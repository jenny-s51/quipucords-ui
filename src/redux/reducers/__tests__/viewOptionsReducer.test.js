import viewOptionsReducer from '../viewOptionsReducer';
import {
  viewTypes,
  viewPaginationTypes,
  viewToolbarTypes,
  credentialsTypes,
  scansTypes,
  sourcesTypes
} from '../../constants';
import { reduxHelpers } from '../../common/reduxHelpers';

describe('viewOptionsReducer', () => {
  it('should return the initial state', () => {
    expect(viewOptionsReducer.initialState).toBeDefined();
  });

  it('should handle specific defined types', () => {
    const specificTypes = [
      viewToolbarTypes.SET_FILTER_TYPE,
      viewToolbarTypes.SET_FILTER_VALUE,
      viewToolbarTypes.ADD_FILTER,
      viewToolbarTypes.REMOVE_FILTER,
      viewToolbarTypes.CLEAR_FILTERS,
      viewToolbarTypes.SET_SORT_TYPE,
      viewToolbarTypes.TOGGLE_SORT_ASCENDING,
      viewPaginationTypes.VIEW_PAGE,
      viewPaginationTypes.SET_PER_PAGE,
      viewTypes.SELECT_ITEM,
      viewTypes.DESELECT_ITEM,
      viewTypes.EXPAND_ITEM
    ];

    specificTypes.forEach(value => {
      const dispatched = {
        type: value,
        viewType: viewTypes.SOURCES_VIEW
      };

      const resultState = viewOptionsReducer(undefined, dispatched);

      expect({ type: value, result: resultState }).toMatchSnapshot(`defined type ${value}`);
    });
  });

  it('should handle all defined fulfilled types', () => {
    const specificTypes = [credentialsTypes.GET_CREDENTIALS, sourcesTypes.GET_SOURCES, scansTypes.GET_SCANS];

    specificTypes.forEach(value => {
      const dispatched = {
        type: reduxHelpers.FULFILLED_ACTION(value),
        payload: {
          data: {
            test: 'success'
          }
        }
      };

      const resultState = viewOptionsReducer(undefined, dispatched);

      expect({ type: reduxHelpers.FULFILLED_ACTION(value), result: resultState }).toMatchSnapshot(
        `fulfilled types ${value}`
      );
    });
  });
});
