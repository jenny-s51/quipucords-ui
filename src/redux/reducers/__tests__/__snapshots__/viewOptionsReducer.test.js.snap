// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viewOptionsReducer should handle all defined fulfilled types: fulfilled types GET_CREDENTIALS 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": undefined,
      "totalPages": NaN,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "GET_CREDENTIALS_FULFILLED",
}
`;

exports[`viewOptionsReducer should handle all defined fulfilled types: fulfilled types GET_SCANS 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": undefined,
      "totalPages": NaN,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "GET_SCANS_FULFILLED",
}
`;

exports[`viewOptionsReducer should handle all defined fulfilled types: fulfilled types GET_SOURCES 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": undefined,
      "totalPages": NaN,
    },
  },
  "type": "GET_SOURCES_FULFILLED",
}
`;

exports[`viewOptionsReducer should handle specific defined types: defined type ADD_FILTER 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [
        undefined,
      ],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "ADD_FILTER",
}
`;

exports[`viewOptionsReducer should handle specific defined types: defined type CLEAR_FILTERS 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "CLEAR_FILTERS",
}
`;

exports[`viewOptionsReducer should handle specific defined types: defined type DESELECT_ITEM 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "DESELECT_ITEM",
}
`;

exports[`viewOptionsReducer should handle specific defined types: defined type EXPAND_ITEM 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "EXPAND_ITEM",
}
`;

exports[`viewOptionsReducer should handle specific defined types: defined type REMOVE_FILTER 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "REMOVE_FILTER",
}
`;

exports[`viewOptionsReducer should handle specific defined types: defined type SELECT_ITEM 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [
        undefined,
      ],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "SELECT_ITEM",
}
`;

exports[`viewOptionsReducer should handle specific defined types: defined type SET_FILTER_TYPE 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": undefined,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "SET_FILTER_TYPE",
}
`;

exports[`viewOptionsReducer should handle specific defined types: defined type SET_FILTER_VALUE 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": undefined,
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "SET_FILTER_VALUE",
}
`;

exports[`viewOptionsReducer should handle specific defined types: defined type SET_PER_PAGE 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": undefined,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "SET_PER_PAGE",
}
`;

exports[`viewOptionsReducer should handle specific defined types: defined type SET_SORT_TYPE 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": undefined,
      "sortType": undefined,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "SET_SORT_TYPE",
}
`;

exports[`viewOptionsReducer should handle specific defined types: defined type TOGGLE_SORT_ASCENDING 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": false,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "TOGGLE_SORT_ASCENDING",
}
`;

exports[`viewOptionsReducer should handle specific defined types: defined type VIEW_PAGE 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": undefined,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 10,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "VIEW_PAGE",
}
`;
